domain/modelconfig/service/service.go

```diff
@@ -280,11 +348,20 @@ func (s *Service) UpdateModelConfig(
 		return fmt.Errorf("making updated model configuration: %w", err)
 	}
 
-	_, err = s.updateModelConfigValidator().Validate(ctx, newCfg, currCfg)
+	modelUUID, modelType, err := s.st.GetModelInfo(ctx)
+	if err != nil {
+		return errors.Annotate(err, "cannot load model info")
+	}
+
+	_, err = s.updateModelConfigValidator(modelType, additionalValidators...).Validate(ctx, newCfg, currCfg)
 	if err != nil {
 		return fmt.Errorf("validating updated model configuration: %w", err)
 	}
 
+	if err := s.runOnSaveHandlers(ctx, modelUUID, updates, removeAttrs); err != nil {
```

Comment:
If you fail after this, you are in an inconsistent state. What is the strategy for notifying the user and/or handling?

Reply:
Yes, that is true. I need to implement a rollback. I wish we had things like 2 phase commit :-(